main/lsp/protocol.cc

```diff
@@ -384,13 +454,8 @@ void cancelRequest(std::deque<std::unique_ptr<LSPMessage>> &pendingRequests, con
     // and ignore.
 }
 
-void LSPLoop::enqueueRequest(const shared_ptr<spd::logger> &logger, LSPLoop::QueueState &state,
-                             std::unique_ptr<LSPMessage> msg, bool collectThreadCounters) {
-    Timer timeit(logger, "enqueueRequest");
-    msg->counter = state.requestCounter++;
-    msg->startTracers.push_back(timeit.getFlowEdge());
-    msg->timers.push_back(make_unique<Timer>(logger, "processing_time"));
-
+void LSPLoop::preprocessAndEnqueue(const shared_ptr<spd::logger> &logger, LSPLoop::QueueState &state,
```

Comment:
The `preprocessingThread` now merges counters, delurking that functionality from this helper method. Otherwise, this is the same method as before, but with a more appropriate name.